File: sdk/src/main/java/com/libreshockwave/chunks/RawChunk.java
package com.libreshockwave.chunks;

import com.libreshockwave.DirectorFile;
import com.libreshockwave.format.ChunkType;
import com.libreshockwave.id.ChunkId;

/**
 * Raw chunk containing unparsed binary data.
 * Used for chunks that haven't been implemented yet.
 */
public record RawChunk(
    DirectorFile file,
    ChunkId id,
    ChunkType type,
    byte[] data
) implements Chunk {

    public int length() {
        return data.length;
    }
}

File: cpp/src/chunks/RawChunk.cpp
#include "libreshockwave/chunks/RawChunk.hpp"

namespace libreshockwave::chunks {

RawChunk::RawChunk(const DirectorFile* file, id::ChunkId id, format::ChunkType type, std::vector<std::uint8_t> data)
    : file_(file), id_(id), type_(type), data_(std::move(data)) {}

const DirectorFile* RawChunk::file() const { return file_; }
format::ChunkType RawChunk::type() const { return type_; }
id::ChunkId RawChunk::id() const { return id_; }
const std::vector<std::uint8_t>& RawChunk::data() const { return data_; }
int RawChunk::length() const { return static_cast<int>(data_.size()); }

} // namespace libreshockwave::chunks
